department-of-veterans-affairs/caseflow
# Justin/appeals 10135

Resolves [APPEALS-10135](https://vajira.max.gov/browse/APPEALS-10135)

### Description
- Created an Admin page for Admin functionality to be built upon
- Created initial folder structure and routing for Admin users
- Only admin users can access, other users will be redirected to /unauthorized
- Created a link within the Misc. tab of the switch users page
- Created a feature toggle for the admin page itself
- Added the feature toggle to [feature flag wiki](https://github.com/department-of-veterans-affairs/caseflow/wiki/Adding-a-Feature-Flag-with-FeatureToggle#currently-implemented-toggles)

### Acceptance Criteria

1. User is required to be a System Admin
2. When not a System Admin if a user tries to go to page directly they are redirected back to root
3. This has its own individual feature toggle that when enabled disables the page from loading ~and link in AC3 from showing~
4. When disabled if a user tries to go to page directly they are redirected back to root
5. Feature toggle is added to https://github.com/department-of-veterans-affairs/caseflow/wiki/Adding-a-Feature-Flag-with-FeatureToggle#currently-implemented-toggles with a state of "In development"
6. Prefix of toggle should be "sys_admin_"
7. The page exists at /admin
8. The Miscellaneous tab on test/users contains a link to the page 
9. There is a h1 header at top of page stating "System Admin UI" above a div with a border
10. This div will be where components for the page are rendered
11. This exists in a new folder called "admin" at https://github.com/department-of-veterans-affairs/caseflow/tree/master/client/app
12. Folder contains an index.jsx file that returns the BrowserRouter component
13. Navigation bar has text "System Admin"
14. https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/components/NavigationBar.jsx
15. <AppFrame><AppSegement> elements contain a PageRoute element for new page
16. The index.js is loaded into https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/index.js
17. https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/index.js is updated to accept new route
18. A pages folder where the page lives is created, the page goes in here
19. An actions/index.jsx, constants.jsx, and reducers/index.js are created 

### Testing Plan
1. Log in as a non-admin user and attempt to access the /admin page. User should be taken to /unauthorized
2. Log in as an admin user and attempt to access the /admin page. User should be taken to /admin

### User Facing Changes
 - ![image](https://user-images.githubusercontent.com/95875751/197601285-aab48667-deb6-47a3-b225-84472d03b996.png)
